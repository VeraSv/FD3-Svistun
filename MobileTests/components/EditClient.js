import React from 'react';
import PropTypes from 'prop-types';
import './EditCard.css'
import {events} from './events';
class EditClient extends React.PureComponent{

  static propTypes = {
    client: PropTypes.object
    } 
    
    state = {
        name:this.props.client.name,
        surname:this.props.client.surname,
        middleName:this.props.client.middleName,
        balance: this.props.client.balance,
        newCard:{id:this.props.client.id, name:this.props.client.name, surname:this.props.client.surname,middleName:this.props.client.middleName, balance:this.props.client.balance},
    }
   
      newSurName=null;
      newName=null;
      newMiddleName=null;
      newBalance=null;
      setNewSurName=(ref)=>{
      this.newSurName=ref
      
      }
      setNewName=(ref)=>{
        this.newName=ref
        
        }
        setMiddleName=(ref)=>{
            this.newMiddleName=ref
            
            }
            setBalance=(ref)=>{
                this.newBalance=ref
                
                }
      setNewText=()=>{
if(this.newSurName) {
    let newText=this.newSurName.value;
      this.setState({surname:newText});
       
}
if(this.newName) {
    let newText=this.newName.value;
      this.setState({name:newText});
       
}
if(this.newMiddleName) {
    let newText=this.newMiddleName.value;
      this.setState({middleName:newText});
       
}
if(this.newBalance) {
    let newText=this.newSurName.value;
      this.setState({balance:newText});
       
}
var newCard=this.state.newCard;
      newCard.surname=this.newSurName.value;
      newCard.name=this.newName.value;
      newCard.middleName=this.newMiddleName.value;
      newCard.balance=this.newBalance.value;
      this.setState({newCard:newCard})
events.emit('EditCard',this.state.newCard);
events.emit('Cancel');
      }

      cancel=()=>{
        events.emit('Cancel');
      }
       render(){
        console.log("MobileClient id=Edit "+this.props.client.name+" render");
          return(   <div><h2>{'Редактирование'}</h2>
         <table className='Card' id={this.props.client.id}>
           <tbody>
           <tr><td>{'Фамилия: '}</td><td><input type={'text'} defaultValue={this.state.surname}  ref={this.setNewSurName}/></td></tr>
             <tr><td>{'Имя: '}</td><td><input type={'text'} defaultValue={this.state.name} ref={this.setNewName}/></td></tr>
             <tr><td>{'Отчество: '}</td><td><input type={'text'} defaultValue={this.state.middleName} ref={this.setMiddleName}/></td></tr>
             <tr><td>{'Баланс: '}</td><td><input type={'text'} defaultValue={this.state.balance}  ref={this.setBalance}/></td></tr>
           </tbody>
         </table>
         <input type={'button'} value={'Save'} className='SaveEdit' onClick={this.setNewText} />
         <input type={'button'} value={'Cancel'} className='CancelEdit' onClick={this.cancel}/>
         </div>)
        
       }
    }
    export default EditClient