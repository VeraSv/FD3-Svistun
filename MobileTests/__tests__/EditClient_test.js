"use strict";

import React from 'react';
import renderer from 'react-test-renderer';

import EditClient from '../components/EditClient.js';

test('работа EditClient', () => {
  let client= {id:1,name:'Иван', surname:'Иванов', middleName:'Иванович', balance:200}

    const component = renderer.create(
        <EditClient client={client} />
      );

      let componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

 

  const buttonCancelEdit = component.root.find((el) =>
  el.type == 'input' && el.props.className == 'CancelEdit');
  buttonCancelEdit.props.onClick();

  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

  buttonCancelEdit.props.onClick();
 

  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();
})