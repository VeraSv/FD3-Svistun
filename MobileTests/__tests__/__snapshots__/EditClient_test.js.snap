// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`работа EditClient 1`] = `
<div>
  <h2>
    Редактирование
  </h2>
  <table
    className="Card"
    id={1}
  >
    <tbody>
      <tr>
        <td>
          Фамилия: 
        </td>
        <td>
          <input
            defaultValue="Иванов"
            type="text"
          />
        </td>
      </tr>
      <tr>
        <td>
          Имя: 
        </td>
        <td>
          <input
            defaultValue="Иван"
            type="text"
          />
        </td>
      </tr>
      <tr>
        <td>
          Отчество: 
        </td>
        <td>
          <input
            defaultValue="Иванович"
            type="text"
          />
        </td>
      </tr>
      <tr>
        <td>
          Баланс: 
        </td>
        <td>
          <input
            defaultValue={200}
            type="text"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <input
    className="SaveEdit"
    onClick={[Function]}
    type="button"
    value="Save"
  />
  <input
    className="CancelEdit"
    onClick={[Function]}
    type="button"
    value="Cancel"
  />
</div>
`;

exports[`работа EditClient 2`] = `
<div>
  <h2>
    Редактирование
  </h2>
  <table
    className="Card"
    id={1}
  >
    <tbody>
      <tr>
        <td>
          Фамилия: 
        </td>
        <td>
          <input
            defaultValue="Иванов"
            type="text"
          />
        </td>
      </tr>
      <tr>
        <td>
          Имя: 
        </td>
        <td>
          <input
            defaultValue="Иван"
            type="text"
          />
        </td>
      </tr>
      <tr>
        <td>
          Отчество: 
        </td>
        <td>
          <input
            defaultValue="Иванович"
            type="text"
          />
        </td>
      </tr>
      <tr>
        <td>
          Баланс: 
        </td>
        <td>
          <input
            defaultValue={200}
            type="text"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <input
    className="SaveEdit"
    onClick={[Function]}
    type="button"
    value="Save"
  />
  <input
    className="CancelEdit"
    onClick={[Function]}
    type="button"
    value="Cancel"
  />
</div>
`;

exports[`работа EditClient 3`] = `
<div>
  <h2>
    Редактирование
  </h2>
  <table
    className="Card"
    id={1}
  >
    <tbody>
      <tr>
        <td>
          Фамилия: 
        </td>
        <td>
          <input
            defaultValue="Иванов"
            type="text"
          />
        </td>
      </tr>
      <tr>
        <td>
          Имя: 
        </td>
        <td>
          <input
            defaultValue="Иван"
            type="text"
          />
        </td>
      </tr>
      <tr>
        <td>
          Отчество: 
        </td>
        <td>
          <input
            defaultValue="Иванович"
            type="text"
          />
        </td>
      </tr>
      <tr>
        <td>
          Баланс: 
        </td>
        <td>
          <input
            defaultValue={200}
            type="text"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <input
    className="SaveEdit"
    onClick={[Function]}
    type="button"
    value="Save"
  />
  <input
    className="CancelEdit"
    onClick={[Function]}
    type="button"
    value="Cancel"
  />
</div>
`;
